madara/crates/client/block_production/src/lib.rs

```diff
@@ -572,6 +627,24 @@ impl<Mempool: MempoolProvider> BlockProductionTask<Mempool> {
                         }
                     }
                 },
+                _ = async {
```

Comment:
i would have written this as
`Some(_) = OptionFuture::from(trigger.as_mut().map(|t| t.notified())`